views/js/runner/plugins/tools/answerMasking/plugin.js

```diff
@@ -0,0 +1,229 @@
+], function ($, _, __, pluginFactory, hider, shortcut, namespaceHelper, answerMaskingFactory) {
+    'use strict';
+
+    /**
+     * The public name of the plugin
+     * @type {String}
+     */
+    var pluginName = 'answerMasking';
+
+    /**
+     * The prefix of actions triggered through the event loop
+     * @type {String}
+     */
+    var actionPrefix = 'tool-' + pluginName + '-';
+
+    /**
+     * Stores the masking state for each item in the test
+     * @type {Object}
+     */
+    var itemStates = {};
+
+    /**
+     * Default Configuration
+     */
+    var defaultConfig = {
+        restoreStateOnToggle: true,
+        restoreStateOnMove: true
+    };
+
+    /**
+     * Returns the configured plugin
+     */
+    return pluginFactory({
+
+        name: pluginName,
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init: function init() {
+            var self = this;
+
+            var testRunner = this.getTestRunner(),
+                testData = testRunner.getTestData() || {},
+                testConfig = testData.config || {},
+                pluginConfig = _.defaults((testConfig.plugins || {})['answer-masking'] || {}, defaultConfig),
+                pluginShortcuts = (testConfig.shortcuts || {})['answer-masking'] || {},
+                $contentArea = this.getAreaBroker().getContentArea();
+
+            var answerMasking = answerMaskingFactory($contentArea);
+
+            function isPluginEnabled() {
+                var context = testRunner.getTestContext() || {},
+                    options = context.options || {};
+                //to be activated with the special category x-tao-option-answerMasking
+                return options.answerMasking && itemContainsChoiceInteraction();
+            }
+
+            function itemContainsChoiceInteraction() {
+                var $container = self.getAreaBroker().getContentArea().parent();
+                return $container.find('.qti-choiceInteraction').length;
+            }
+
+            function togglePluginButton() {
+                if (isPluginEnabled()) {
+                    self.show();
+                } else {
+                    self.hide();
+                }
+            }
+
+            function togglePlugin() {
+                if (! answerMasking.getState('enabled')) {
+                    enableMasking();
+                } else {
+                    disableMasking();
+                }
+            }
+
+            function enableMasking() {
+                var testContext = testRunner.getTestContext(),
+                    itemId = testContext.itemIdentifier;
+
+                answerMasking.enable();
+                if (pluginConfig.restoreStateOnToggle) {
+                    answerMasking.setMasksState(itemStates[itemId]);
+                }
+                self.button.turnOn();
+                testRunner.trigger('plugin-start.answer-masking');
```

Comment:
Same remark as for the config, you should use the plugin name for more consistency.
You could also use the `plugin.trigger()` method that will automatically prefix the event.